widgetastic_manageiq/__init__.py

```diff
@@ -5024,6 +5042,43 @@ def is_displayed(self):
         return self.browser.is_displayed(self)
 
 
+class ConversionHostProgress(Widget):
+    """Represents the progress widget for conversion Host configuration"""
+
+    ROOT = './/div[contains(@id,"conversion_hosts")]'
+
+    ITEM_LOCATOR = './/div[contains(@class,"list-view-pf-main-info")]'
+    ERROR_LOCATOR = './/div/span[contains(@class,"pficon-error-circle-o")]'
+    OK_LOCATOR = './/div/span[contains(@class,"pficon-ok")]'
+
+    SPINNER_LOCATOR = './/div[contains(@class,"spinner")]'
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def _conversion_host(self, hostname):
+        for el in self.browser.elements(self.ITEM_LOCATOR):
+            if hostname in self.browser.text(el):
+                return el
+        raise ItemNotFound("No host found with host name : {}".format(hostname))
+
+    def in_progress(self, hostname):
+        el = self._conversion_host(hostname)
+        return self.browser.is_displayed(self.SPINNER_LOCATOR, parent=el)
+
+    def is_host_configured(self, hostname):
+        """Returns True if OK is displayed and spinner not present"""
+        el = self._conversion_host(hostname)
+        error_displayed = self.browser.is_displayed(self.ERROR_LOCATOR, parent=el)
+        ok_displayed = self.browser.is_displayed(self.OK_LOCATOR, parent=el)
+        if ok_displayed:
```

Comment:
Why not just have `return okay_displayed or not error_displayed`? Or just `return okay_displayed`? Why is the `error_displayed` necessary?

Reply:
Good point :+1: